### PR TITLE
[SYCL][Doc] Add release notes for September release

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

File: sycl/ReleaseNotes.md [hunk 1]
```diff
@@ -1,3 +1,246 @@
...
+   explicit, linking against `lib/libsycl-fallback-cassert.o` or
+   `lib/libsycl-fallback-cassert.spv`
+ - [new] Performance may be impacted by JIT-ing an extra 'copier' kernel and due
+   running the 'copier' kernel and host-task after each kernel which uses
+   assert
+ - [new] Driver issue. When a two-step AOT build is used and there's at least a
+   single call to devicelib function from within kernel, the device binary
+   image gets corrupted
+ - [new] Filenames are removed from diagnostic to avoid leaks of full paths
+   from build environment [6bbfe42b3]
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:208]: I don't think it's an issue. It's a change and it's a fix.

**Comment by AaronBallman** [comment]
Should our release notes attempt to convey which Clang revision we are most closely a downstream of? e.g., so users know this is more closely related to future-Clang 14 vs current-Clang 13. (Giving users this information helps them to better understand how much of Clang's features they can expect to work here.)

File: sycl/ReleaseNotes.md [hunk 2]
```diff
@@ -1,3 +1,246 @@
...
+   in host tasks [f823d61]
+ - Fixed a crash which could happen when submitting tasks to multiple queues
+   from multiple threads [4ccfd21]
+ - Fixed a possible multiple definitions problem which could happen when using
+   `fsycl-device-code-split` option [d21082f]
+ - Fixed several problem in configuration file processing [2a35df0]
+ - Fixed imbalance in events release and retain in Level Zero plugin [6117a1b]
+
+## API/ABI breakages
+
```
[the remark this answers is not in the document]
Comment (romanovvlad) [reply sycl/ReleaseNotes.md:195]: Funny enough, but massive API "removals" didn't get into the commits range for the release. So, it seems we have only one API breakage here.

File: sycl/ReleaseNotes.md (hunk 1 above)
Comment (romanovvlad) [reply sycl/ReleaseNotes.md:208]: Moved to bug fixes section.

**Comment by romanovvlad** [comment]
> 
> 
> Should our release notes attempt to convey which Clang revision we are most closely a downstream of? e.g., so users know this is more closely related to future-Clang 14 vs current-Clang 13. (Giving users this information helps them to better understand how much of Clang's features they can expect to work here.)

Suggestion is to include the hash of the latest `llvm.org` commit that was in `intel/llvm` , right?
For example, for this release we would have `83ddfa0`.

**Review by pvchupin — APPROVED** [review]

File: sycl/ReleaseNotes.md [hunk 3]
```diff
@@ -1,3 +1,246 @@
+# September'21 release notes
+
+Release notes for commit range 4fc5ebe..bd68232
+
+## New features
+### SYCL Compiler
+ - Disabled kernel function propagating up the call tree to callee when in
+   SYCL 2020 mode [2667e3e]
```
Comment (AlexeySachkov) [inline sycl/ReleaseNotes.md:8]: I wouldn't call it a new feature. To me it looks like a bug fix. If anything, we have removed functionality from the compiler  in that patch

File: sycl/ReleaseNotes.md [hunk 4]
```diff
@@ -1,3 +1,246 @@
...
+ - Added default device triple `spir64` when the compiler encounters any
+   incoming object/libraries that have been built with the `spir64` target.
+   `-fno-sycl-link-spirv` can be used for disabling this behaviour [1342360]
+ - Added support for non-uniform `IMul` and `FMul` operation for `ptx-nvidiacl`
+   [98a339d]
+ - Added splitting modules capability when compiling for NVPTX and AMDGCN
+   [c1324e6]
+ - Added `-fsycl-footer-path=<path>` command-line option to set path where to
+   store integration footer [155acd1]
+ - Improved read only accessor handling - add `readonly` attribute to the
```
Comment (AlexeySachkov) [inline sycl/ReleaseNotes.md:48]:
```suggestion
 - Improved read only accessor handling - added `readonly` attribute to the
```

File: sycl/ReleaseNotes.md [hunk 5]
```diff
@@ -1,3 +1,246 @@
...
+ - Removed faulty implementations for atomic loads and stores for `acquire`,
+   `release` and `seq_cst` memory orders libclc for NVPTX [4876443]
+ - Fixed having two specialization for the `specialization_id`, one of which was
+   invalid [f71a1d5]
+ - Fixed context destruction in HIP plugin [6042d3a]
+ - Changed `queue::mem_advise` and `handler::mem_advise` to take `int` instead
+   of `pi_mem_advice` [af2bf96]
+ - Prevented passing of `-fopenmp-simd` to device compilation when used along
+   with `-fsycl` [226ed8b]
+ - Correctly form the integration header when non-base-ascii chars are used in
```
Comment (AlexeySachkov) [inline sycl/ReleaseNotes.md:144]:
```suggestion
 - Fixed generation of the integration header when non-base-ascii chars are used in
```

File: sycl/ReleaseNotes.md [hunk 6]
```diff
@@ -1,3 +1,246 @@
...
+   the kernel name [91f50477]
+ - Fixed a problem which could lead to picking up incorrect kernel at runtime in
+   some situations when unnamed lambda feature is used [27c632e]
+ - Fixed suggested target triple in the warning message [7cc89fa]
+ - Fix identity for multiplication on CUDA backend [a6447ca4]
```
Comment (AlexeySachkov) [inline sycl/ReleaseNotes.md:149]:
```suggestion
 - Fixed identity for multiplication on CUDA backend [a6447ca4]
```

File: sycl/ReleaseNotes.md (hunk 3 above)
Comment (romanovvlad) [reply sycl/ReleaseNotes.md:8]: Agree, but it doesn't sound like a bug fix as well. Does "improvements" section sound good for this for you?

Comment (AlexeySachkov) [reply sycl/ReleaseNotes.md:8]: > Does "improvements" section sound good for this for you?

Yeah, that sounds better, i.e. we are one step closer to be compliant with SYCL 2020 spec

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**